### ISSUE TITLE
[Balance] Gatling Barrels were nerfed the wrong way (and more)

### ISSUE THREAD

**Issue by SabreMania** [tsunamayo/Skywanderers#1715]
Greetings,

After this new update, many things were changed in rather drastic ways. For starters, the gatling barrel was 'nerfed' by being given a very strange burst mode. This burst mode is forced upon the gun, it cannot be toggled on or off, and the amount of time you fire is the same as the amount of time you have to wait until your next burst.

And on top of it, the damage per shot is still the same, which was the issue that the previous version's gatling had. So first:

-**Dispersion for all guns is far too big**, even when your barrel is maxed out in length. The only way to hope to actually hit a target at a range is to spam guns and spam turrets, and hope you hit your mark. This is especially true for the gatling barrel. If this is what 5% dispersion looks like, I would suggest to put it much further down, perhaps at around **1% or below.**

-**The gatling barrel needs it's continous fire mode back**, at the very least for brick gatling barrels. If they don't, not only will it seriously impact PD performance, it will negatively affect all fighters.

-**The Burst mode** in and of itself is a good concept, but the way it is implemented needs to change. **It needs to be something that can be toggled on or off**, and it needs the player to be directly able to choose the length of the burst, or at the very least, how long the wait time between each burst is. 

-**The damage needs to be lowered.** This is more of a 'for the future' kind of balancing problem, but the previous gatling barrel's problem was that it did as much damage as all of the others, and could fire faster than all of the others. This is still true, though the dispersion negates that aspect right now. However, after dispersion is fixed, this will become an issue again, and lowered damage is how it should have been dealt with in the first place.

My next complaint would be surrounding the new add-ons (as well as report a bug).

-For a strange reason I have not been able to place down brick add ons on brick weapons. (But I have been able to place a laser and a beam unit where the addons for brick weapons is, which was... very wrong and very probably unwanted)

-The add ons themselves are... pretty big. I will admit that me and at least a few others (I have not asked the community as a whole) were not expecting them to be like that. We were expecting them to, in a way, complete the cylindrical shape that the base weapons have, but instead we have these huge canisters that are plopped on top of the guns. This makes it very hard to make them fit inside of a smaller turret, and forces turrets to be unarmored if they are below a certain size. (They also look kind of bad, when you imagine a cannon you generally don't see big pillars standing up from the breech)

I'm not sure how to conclude this post, so I will leave it at that. Anyone can feel free to add onto this, and or criticize my points.

**Comment by ultrak2k** [comment]
The addons do look pretty wacky as of now, though the cylnders in my opinon are okay, and you can't rotate the triangular one, and I haven't personally had the bug, though I only used brick beams, however I completelty agree with the rest of these points. The gatling could have been nerfed by decreasing its damage overall, no? Most other games with a similar system nerf rate of fire with rapid fire weapons to balance them out - a burst weapon is cool, but it should either be a module or a barrel, instead of a forced thing for gatling.

**Comment by tsunamayo** [comment]
I see, the big problem was to have a gatling compatible with the new side addon.
In this case with a continuous mode what the overcharge (cylinder) and supercharge (the triangle) will do?
Fow now:
If you put a lot of supercharge you can reduce the cooldown time.
If you put overcharge you have a bigger burst (but longer charge time).
I could switch to continuous fire if we find something meaningful for both addons.

Spread was reduced to 2deg, I will tweak again if necessary.
For the look I am not sure, I cant make them too small either. I am waiting for more input on that - but for sure it is always gonna be gameplay vs look.
Also for the future, I plan to have 3 weapons tier for weapon progression: 1 row addon, 2 row addon and last tier will be 3. So 3 row of supercharge with the gatling will give continuous fire.
Cheers

**Comment by SabreMania** [comment]
I am unable to comment on how to make the side addons compatible with the gatling, as they are too new and I haven't experimented all that much with them. Reading over their description makes them seem a bit confusing too at the moment.

With that being said, what I do know is that making the gatling barrel be forced to have a burst mode is bad and unintuitive, if there is no option for an automatic firing weapon. 

Having 3 rows of supercharge to make a continuous fire gun might sound good, but I fear that it will end up being the only thing that will be used for the gatling. Fighters need to be able to start and stop firing whenever the pilot wants/needs them to, without an artificial cooldown, and forcing them to pick this specific combinaison of gatling + addon downplays the aspect of having addons in the first place, at least for fighters.

**Comment by ultrak2k** [comment]
> I see, the big problem was to have a gatling compatible with the new side addon.
> In this case with a continuous mode what the overcharge (cylinder) and supercharge (the triangle) will do?
> Fow now:
> If you put a lot of supercharge you can reduce the cooldown time.
> If you put overcharge you have a bigger burst (but longer charge time).
> I could switch to continuous fire if we find something meaningful for both addons.
> 
> Spread was reduced to 2deg, I will tweak again if necessary.
> For the look I am not sure, I cant make them too small either. I am waiting for more input on that - but for sure it is always gonna be gameplay vs look.
> Also for the future, I plan to have 3 weapons tier for weapon progression: 1 row addon, 2 row addon and last tier will be 3. So 3 row of supercharge with the gatling will give continuous fire.
> Cheers

I think overcharge should turn the weapon into a burst fire - extreme rate of fire for a short period of time, non continuing, while supercharge should perhaps increase fire rate HOWEVER also increase dispersion for Gatling.

**Comment by tsunamayo** [comment]
**Overcharge** => basically store more power into the weapon. Beam fire longer and laser fire in volley. So if you do a lot of hit and run instead of firing continuously this is basically becomes free DPS, as you get to unleash more while it cost you the same. Also if you do some system sniping, having two shot close together will maximize chance of penetration. 
**Supercharge** => Accelerate the charging rate of the weapon. Came at a cost for regen.
You can suggest new name if you find that confusing. 

Okay, also please also keep in mind that the gatling is already firing quite fast. I cant increase the number of shot too much, this would stress network and the game perfs.
Lets continue brainstorming about that, see if other people have idea.
My initial fear is that everybody wants to use gatling. I have to find something that satisfy player idea and fits in the system...

**Comment by SabreMania** [comment]
The reason why everyone wants, or would want, to use the gatling is because it's damage per shot is the same as the other barrels, but it fires much faster than the others. If you reduce it's damage per shot, it will make other barrels more valuable.

I understand the concerns over game performance. However the increase in fire rate doesn't need to be that significant. Maybe a 3 or 5% increase, maybe a little more, just enough to be attractive?

If this is still not acceptable, then I am not sure how Overcharge could affect a full-auto gatling

**Comment by tsunamayo** [comment]
Yeah supercharge could be simply making shot more powerful, which would make sense as there is no load time. Overcharge I am not sure, the spirit of it is it does not cost anything, it just "store" power before. Could be a more powerfull shot for x seconds then normal shot when empty...

**Comment by SabreMania** [comment]
Like we said, Overcharge could turn the gatling into a burst weapon, specifically in the way that it works right now, though perhaps with a bit more shots per burst?

**Comment by ultrak2k** [comment]
> Yeah supercharge could be simply making shot more powerful, which would make sense as there is no load time. Overcharge I am not sure, the spirit of it is it does not cost anything, it just "store" power before. Could be a more powerfull shot for x seconds then normal shot when empty...

Right now, isn't gatling the same range as all other weapons? I also think the gatling range should be reduced, having a midrange assault weapon is strange and also contributed to unbalance with gats.

**Comment by tsunamayo** [comment]
Range should be reduced, as range and velocity are the same (and velocity is reduced), laser live for 3 seconds. So they travel less during that time.

**Comment by SabreMania** [comment]
@tsunamayo Alright, I have tested the gatling barrel again post-update, with 2.5 degree accuracy. The barrel is maxed out in length, and this was the result
![spreadgat](https://user-images.githubusercontent.com/52141971/79039289-3d3aa180-7bae-11ea-8dd6-dac718824944.png)

Keep in mind this is not a test to see the damage output in a given frame of time, it is merely to test how wide a 2.5 degree spread means.

The first panel is 100 meters away from the barrel's tip, and the damage zone formed a 10 by 10 square.

The second pannel is 200 meters away, and the damage zone formed an 18 by 18 square.

The third pannel is 500 meters away, and the damage zone formed a 43 by 43 square.

With such results, me, and many others, believe that the Gatling barrel still needs an improvement in accuracy. However, how much is questionable. An important question arises, namely:

What do you have in mind for the gatling barrel? Do you want it to be usable as a point defense gun capable of intercepting missiles as they approach? Do you want it to be used as an anti fighter weapon on ship turrets? Or are they strictly meant to be a close range fighter vs fighter weapon, as well as a close range 'brawler' kind of damage dealer for big ships against other big ships?

**Comment by tsunamayo** [comment]
@SabreMania yes thanks for your analysis, indeed it is a lot of dispersion but again I prefer to start on too much side than the opposite. The only advandage of other barrel is the precision, so if the gatling was reasonably okay to use in many cases then whats the point of the other barrel?
My philosophy is Blizzard like, exaggerate units trait and then reduce it after.
One of the traits of the gatling must be very lack of precision, and it is just the base rate, with can be reduced to almost the normal spread with the very long barrel.
I dont see this as a anti-fighter at all, more at a close range assault pure DPS weapon, to chew enemy hull. But again I am waiting to see how the meta develop, things will change.
For anti-fighter / anti-missile I had in mind a flak shell for kinetic gun.
In my plan, the kinetic gun will have the same barrel as the laser, and no side addon.
Instead you need to craft ammo: choose a shell: exposive, flak, penetrative, large shell (big payload but slower) ect, and a or several payload. Should be fun.
As this is building game, my only wish is to have player have fun experimenting different design, and have enough depth so that different approach are possible.

**Comment by SabreMania** [comment]
@tsunamayo Ahhh. This clears up a lot of things actually. I can't speak for everyone else, but I for sure saw the gatling barrel as the go-to option for Point Defense and anti-fighter weaponry. 

While I do like the idea of close-proximity shells, this makes me a little bit concerned. What will ships who desire another option use for anti fighter and point defense duty? If for example I make a ship that doesn't use kinetic, but does uses laser units, will there be an equivalent?

I really like the idea of different shells, it sounds very cool.

But to come back to the gatling barrel. Do you have any plans to make the brick gatling more accurate, and give it a full-auto mode? I feel like people will want to use this specific barrel for their fighters, as there is no other option right now that offers a high rate of fire, and a fighter needs to be able to start and stop shooting when the pilot needs it to.

**Comment by tgiddings** [comment]
A user-experience perspective on this: continuous-fire weapons just feel cool. If a continuous-fire weapon and a burst-fire weapon have the same DPS, I and likely many others will find the continuous-fire weapon a more enjoyable experience most of the time.

Gameplay-wise, a skilled player might be able to achieve a higher average DPS with a burst-fire weapon since they can save the damage for a moment where there's a better chance to hit, while an unskilled player will probably need the continuous-fire weapon's ability to spray a large area in order to get hits in.
